plutus-pab/src/Plutus/PAB/Core/ContractInstance.hs

```diff
@@ -82,16 +119,10 @@ activateContractSTM ::
     => (Eff appBackend ~> IO)
     -> ContractActivationArgs (ContractDef t)
     -> Eff effs ContractInstanceId
-activateContractSTM runAppBackend a@ContractActivationArgs{caID, caWallet} = do
-    activeContractInstanceId <- ContractInstanceId <$> uuidNextRandom
-    logDebug @(ContractInstanceMsg t) $ InitialisingContract caID activeContractInstanceId
-    initialState <- Contract.initialState @t activeContractInstanceId caID
-    Contract.putStartInstance @t a activeContractInstanceId
-    Contract.putState @t a activeContractInstanceId initialState
-    s <- startSTMInstanceThread @t @m runAppBackend a activeContractInstanceId
-    ask >>= void . liftIO . STM.atomically . InstanceState.insertInstance activeContractInstanceId s
-    logInfo @(ContractInstanceMsg t) $ ActivatedContractInstance caID caWallet activeContractInstanceId
-    pure activeContractInstanceId
+activateContractSTM runAppBackend a@ContractActivationArgs{caID} = do
+  activeContractInstanceId <- ContractInstanceId <$> uuidNextRandom
+  initialState <- Contract.initialState @t activeContractInstanceId caID
```

Comment:
this could be extracted into a separate function that returns a `ContractInstanceState t`